pipeline/stage02_processing/scripts/hierarchical_spatial_sampling.py

```diff
@@ -0,0 +1,223 @@
+import neo
+import numpy as np
+from scipy.stats import shapiro
+import matplotlib.pyplot as plt
+import argparse
+from utils.io import load_neo, write_neo, save_plot
+from utils.parse import none_or_float, none_or_int, none_or_str
+from utils.neo import analogsignals_to_imagesequences, imagesequences_to_analogsignals
+
+
+def next_power_of_2(n):
+    if n == 0:
+        return 1
+    if n & (n - 1) == 0:
+        return n
+    while n & (n - 1) > 0:
+        n &= (n - 1)
+    return n << 1
```

Comment:
This function is really hard to read and understand. It could be however just be replaced by 
```
exponent = np.ceil(np.log2(n))
return int(2**exponent)
```